lib/src/util/utf8.cc

```diff
@@ -18,7 +19,15 @@ namespace Util {
                 return false;
             }
         }
-        return true;
+
+        // rapidjson::UTF8<char>::Validate accepts null bytes as valid UTF-8.
+        // They technically are valid since they're the null character. But
+        // null characters aren't valid in a string, so we want to disallow
+        // them regardless.
+        bool has_null = std::any_of(s.begin(), s.end(), [](char c) {
+                            return c == 0;
```

Comment:
Note that the only time a null byte is allowed in UTF-8 is for a single-byte null character. A null byte is _not_ allowed as part of a multibyte character.

https://en.wikipedia.org/wiki/UTF-8#Encoding